src/extension.ts

```diff
@@ -675,25 +675,23 @@ export class Extension implements RunHooks {
           activeDecorations.push({ range: location.range });
       }
 
-      const completedDecorations: vscodeTypes.DecorationOptions[] = [];
+      const completedDecorations: Record<number, vscodeTypes.DecorationOptions> = {};
       for (const { location, duration } of completed) {
         if (uriToPath(location.uri) === uriToPath(editor.document.uri)) {
```

Comment:
drive-by I would pre-compute the editor's uri instead of doing it 1000 times. Maybe even store decorations by line from the get go.